tensornetwork/network_components.py

```diff
@@ -181,17 +181,24 @@ def reorder_edges(self, edge_order: List["Edge"]) -> "BaseNode":
       This node post reordering.
 
     Raises:
-      ValueError: If either the list of  edges is not the same as expected or
+      ValueError: If either the list of edges is not the same as expected or
         if you try to reorder with a trace edge.
       AttributeError: If the Node has no tensor.
 
     """
     if not hasattr(self, '_tensor'):
       raise AttributeError("Please provide a valid tensor for this Node.")
 
-    if set(edge_order) != set(self.edges):
+    extra_edges = set(edge_order).difference(set(self.edges))
+    if extra_edges:
       raise ValueError("Given edge order does not match expected edges. "
-                       "Found: {}, Expected: {}".format(edge_order, self.edges))
+                       "Additional edges that do not belong to node found: "
+                       "{}".format(extra_edges))
```

Comment:
Would it make more sense to print the edge names instead? It would be easier for humans to read, but if the names are conflicting it might become confusing. What is your opinion?

Reply:
Well, it's what I did first, but then I also noticed the conflicting issue :(. 
We can go for the hybrid approach and print both name and object reference:
`Additional edges that do not belong to node found: Edge "my_edge_name" (tensornetwork.network_components.Edge object at 0x1a5c318f50>), Edge "my_second_edge_name' (another hash)`

Reply:
We might be able to do that by changing the `__repr__` method on edges.